pkg/apis/pravega/v1beta1/pravegacluster_types.go

```diff
@@ -1194,6 +1206,66 @@ func (p *PravegaCluster) validateConfigMap() error {
 	return nil
 }
 
+func (p *PravegaCluster) ValidateSegmentStore() error {
+	totalMemoryLimitsQuantity := p.Spec.Pravega.SegmentStoreResources.Limits[corev1.ResourceMemory]
+	totalMemoryRequestsQuantity := p.Spec.Pravega.SegmentStoreResources.Requests[corev1.ResourceMemory]
+	if (resource.Quantity{}) == totalMemoryLimitsQuantity {
+		return fmt.Errorf("Missing required value for field spec.pravega.segmentStoreResources.limits.memory")
+	}
+
+	totalMemoryLimits := totalMemoryLimitsQuantity.Value()
+	totalMemoryRequests := totalMemoryRequestsQuantity.Value()
+	if totalMemoryLimits < totalMemoryRequests {
+		return fmt.Errorf("spec.pravega.segmentStoreResources.requests.memory value must be less than or equal to spec.pravega.segmentStoreResources.limits.memory")
+	}
+
+	cacheSizeString := p.Spec.Pravega.Options["pravegaservice.cache.size.max"]
+	if cacheSizeString == "" {
+		return fmt.Errorf("Missing required value for option pravegaservice.cache.size.max")
+	}
+	cacheSizeQuantity := resource.MustParse(cacheSizeString)
+	maxDirectMemoryString := ""
+	xmxString := ""
+
+	for _, value := range p.Spec.Pravega.SegmentStoreJVMOptions {
+		if strings.Contains(value, "-Xmx") {
+			xmxString = strings.ToUpper(strings.TrimPrefix(value, "-Xmx")) + "i"
+		}
+
+		if strings.Contains(value, "-XX:MaxDirectMemorySize=") {
+			maxDirectMemoryString = strings.ToUpper(strings.TrimPrefix(value, "-XX:MaxDirectMemorySize=")) + "i"
+		}
+	}
+
+	if xmxString == "" {
+		return fmt.Errorf("Missing required value for Segment Store JVM Option -Xmx")
+	}
+	xmxQuantity := resource.MustParse(xmxString)
+
+	if maxDirectMemoryString == "" {
+		return fmt.Errorf("Missing required value for Segment Store JVM option -XX:MaxDirectMemorySize")
+	}
+	maxDirectMemoryQuantity := resource.MustParse(maxDirectMemoryString)
+
+	xmx := xmxQuantity.Value()
+	maxDirectMemorySize := maxDirectMemoryQuantity.Value()
+	cacheSize := cacheSizeQuantity.Value()
+
+	if totalMemoryLimits <= maxDirectMemorySize+xmx {
```

Comment:
Nit: please, leave a space after and before the `+` symbol (it is hard to read otherwise).